ci/post/changelog.py

```diff
@@ -0,0 +1,158 @@
+
+    @param[in] response The response from a GET call
+
+    @returns The page number of the last page, or MAX_PAGES
+    """
+
+    # The link header is a heckn dictionary in string form, and for that matter the
+    # value is in front of the key.  It is of the form '<<url1>>; rel=<key1>, <url2>, <key2>'
+    link = response.headers['link']
+    link = link.split(',')
+    for x in link:
+        y = x.split(';')        # <url>; rel=<key>
+        if "last" not in y[1]:  # look for rel='last'
+            continue
+        # search in the url for `page=`` and grab the int immediately after
+        # the int may be delimited by '&' or '>'
+        str = y[0][y[0].find('page=') : ]   # slice everything out before 'page='
+        str = str.split('&', 1)[0]          # slice out any possible queries after page=<int>
+        last_page = int( re.sub('\D', "", str) )    # strip out just the page number, as integer
+        if last_page > MAX_PAGES:
+            return MAX_PAGES
+        else:
+            return last_page
+    
+
+def main():
+    os.chdir(os.path.dirname(__file__))	# Change working directory to this file's directory
+    
+    start_date = datetime.strptime(sys.argv[1], "%Y-%m-%d")
+
+    print("start_date: {}".format(start_date))
+
+    # Retrive the git log for master, since start date, order by date, in ascending order, use YYYY-MM-DD date format,
+    #   each line is of the form "<author short date> | <message>", only grab commits from master,
+    #   output to `fsofordigest.txt`,
+    
+    # check_output(("git log master", "--after=2021-09-15", "--date-order", "--reverse", "--date=short",
+    # "--pretty=\"%as | %s\"",  "--first-parent",
+    # "--output fsofordigest.txt"), text=True)
+
+    # The commit messages are either of the form "<body> (#<PR_number>)" or "Merge pull request #<PR_number> from <source_repo>"
+    # For our purposes, we want them be all of the form "<body> (#<PR_number>)"
+    # Typically, the second form does not have a usable body, so we'll have to pull the PR title from github
+
+    # Github's REST API allows us to retrieve PR's, but due to it being HTTP requests it has to page in the data.
+    # We can loop over the GET request and check for a merge date.
+    # The GET request doesn't have sorting by merge date, but does have sorting by the last time it was updated... which is usually
+    # also the merge date but not always.
+
+    # Key attributes within the PR JSON objects:
+    #   "merged_at: <date>"
+    #   "merged: <bool>"
+    #   "html_url: <string>"
+    #   "number: <int>"     # The PR id number
+    #   "title: <string>"   # Title of the PR
+    #   "body: <string>"    # The body paragraph detailing the PR
+    #   "labels: [label_object]"    # All labels attached to this PR
+
+    changelog = {}
+    url = "https://api.github.com/repos/scp-fs2open/fs2open.github.com/pulls"
+    page = 1
+    last_page = MAX_PAGES
+    while (page <= last_page):
```

Comment:
You could use `for page in range(1, last_page + 1):` instead if you remove the `page = 1` and `page += 1` lines. It's not a huge difference but IMO a bit clearer.